src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathDocumentBuilder.cs

```diff
@@ -126,22 +133,22 @@ public void Initialize(XPathDocument doc, IXmlLineInfo lineInfo, string baseUri,
         /// <summary>
         /// XPathDocument ignores the DocType information.
         /// </summary>
-        public override void WriteDocType(string name, string pubid, string sysid, string subset)
+        public override void WriteDocType(string name, string? pubid, string? sysid, string? subset)
         {
         }
 
         /// <summary>
         /// Shortcut for calling WriteStartElement with elemType == null.
         /// </summary>
-        public override void WriteStartElement(string prefix, string localName, string ns)
+        public override void WriteStartElement(string? prefix, string localName, string? ns)
         {
             this.WriteStartElement(prefix, localName, ns, string.Empty);
         }
 
         /// <summary>
         /// Build an element node and attach it to its parent, if one exists.  Make the element the new parent node.
         /// </summary>
-        public void WriteStartElement(string prefix, string localName, string ns, string baseUri)
+        public void WriteStartElement(string? prefix, string localName, string? ns, string? baseUri)
```

Comment:
unfortunate assert below... you might consider fixing

Reply:
You mean remove the assert? I think is there to note that regardless of base signature and other implementations that can handle null, this one does not... 

Also, they are passed into `XmlNameTable.Add` which does not accept null.

Reply:
consider `prefix ??= string.Empty; ns ??= string.Empty` - might be worth to see how other implementations handle it. Treat as optional though

Reply:
The impl. of [`XmlTextWriter`](https://github.com/dotnet/runtime/blob/32ddcd3f7056914c95e600c0f2055b632e323463/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs#L485), which is public is capable of handle null values.

Many other impl. also do this pattern of asserting parameters != null:
https://github.com/dotnet/runtime/blob/32ddcd3f7056914c95e600c0f2055b632e323463/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriter.cs#L300-L303

https://github.com/dotnet/runtime/blob/32ddcd3f7056914c95e600c0f2055b632e323463/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriter.cs#L1826-L1828

I would prefer to leave it as is to align with the other implementations.